tests/unit_tests/Fitting/minimizers/test_minimizer_dfo.py

```diff
@@ -0,0 +1,263 @@
+import pytest
+
+from unittest.mock import MagicMock
+import numpy as np
+
+import easyscience.fitting.minimizers.minimizer_dfo
+from easyscience.Objects.new_variable import Parameter
+
+from easyscience.fitting.minimizers.minimizer_dfo import DFO
+from easyscience.fitting.minimizers.utils import FitError
+
+
+class TestDFOFit():
+    @pytest.fixture
+    def minimizer(self) -> DFO:
+        minimizer = DFO(
+            obj='obj',
+            fit_function='fit_function', 
+            method='leastsq'
+        )
+        return minimizer
+
+    def test_init(self, minimizer: DFO) -> None:
+        assert minimizer._p_0 == {}
+        assert minimizer.wrapping == 'dfo'
+
+    def test_init_exception(self) -> None:
+        with pytest.raises(FitError):
+            DFO(
+                obj='obj',
+                fit_function='fit_function', 
+                method='not_leastsq'
+            )
+
+    def test_available_methods(self, minimizer: DFO) -> None:
+        # When Then Expect
+        assert minimizer.available_methods() == ['leastsq']
+
+    def test_fit(self, minimizer: DFO) -> None:
+        # When
+        from easyscience import global_object
+        global_object.stack.enabled = False
+
+        mock_model = MagicMock()
+        mock_model_function = MagicMock(return_value=mock_model)
+        minimizer._make_model = MagicMock(return_value=mock_model_function)
+        minimizer._dfo_fit = MagicMock(return_value='fit')
+        minimizer._set_parameter_fit_result = MagicMock()
+        minimizer._gen_fit_results = MagicMock(return_value='gen_fit_results')
+
+        cached_par = MagicMock()
+        cached_par.raw_value = 1
+        cached_pars = {'mock_parm_1': cached_par}
+        minimizer._cached_pars = cached_pars
+
+        # Then
+        result = minimizer.fit(x=1.0, y=2.0)
+
+        # Expect
+        assert result == 'gen_fit_results'
+        minimizer._dfo_fit.assert_called_once_with(cached_pars, mock_model)
+        minimizer._make_model.assert_called_once_with(parameters=None)
+        minimizer._set_parameter_fit_result.assert_called_once_with('fit', False)
+        minimizer._gen_fit_results.assert_called_once_with('fit', 1.4142135623730951)
+
```

Comment:
isn't the actual fit test done in `tests/integration_tests/Fitting/test_fitter.py` in `test_basic_fit` method?

Reply:
Yes, you can claim that the "real" test is done in the integration test.  In this unit test we have mocked out the functionality outside the method and it just test if the "external" methods are called with the expected arguments